R/get_gene_expression.R

```diff
@@ -77,218 +129,61 @@
-    tidy_expression_file = paste0(base_path, tidy_expression_path)
+  sample_details = filter(sample_details,
+                          mrna_sample_id %in% these_samples_metadata$sample_id | 
+                          capture_sample_id %in% these_samples_metadata$sample_id |
+                          genome_sample_id %in% these_samples_metadata$sample_id)
+  if(verbose){
+    remaining_rows = nrow(sample_details)
+    message(paste(remaining_rows,"samples from your metadata have RNA-seq data available"))
   }
-  
-  if(!missing(expression_data)){
-    tidy_expression_data = as.data.frame(expression_data) #is this necessary? Will it unnecessarily duplicate a large object if it's already a data frame?
-    if(!missing(hugo_symbols)){
-      #lazily filter on the fly to conserve RAM
-      wide_expression_data = tidy_expression_data %>%
-        dplyr::filter(Hugo_Symbol %in% hugo_symbols) %>%
-        dplyr::select(-ensembl_gene_id) %>%
-        group_by(mrna_sample_id,Hugo_Symbol) %>% #deal with non 1:1 mapping of Hugo to Ensembl
-        slice_head() %>%
-        as.data.frame() %>%
-        pivot_wider(names_from = Hugo_Symbol, values_from = expression)
-    }else if(!missing(ensembl_gene_ids)){
-      wide_expression_data = tidy_expression_data %>%
-        dplyr::filter(ensembl_gene_id %in% ensembl_gene_ids) %>%
-        dplyr::select(-Hugo_Symbol) %>%
-        as.data.frame() %>%
-        pivot_wider(names_from = ensembl_gene_id, values_from = expression)
-    }else{
-      
-      #for when a user wants everything. Need to handle the option of getting back Hugo_Symbol instead
-      wide_expression_data = tidy_expression_data %>%
-        dplyr::select(-Hugo_Symbol) %>%
-        as.data.frame() %>%
-        pivot_wider(names_from = ensembl_gene_id, values_from = expression)
-    }
-  }else{
-    if(!file.exists(tidy_expression_file)){
-      print(paste("missing: ", tidy_expression_file))
-      message("Cannot find file locally. If working remotely, perhaps you forgot to load your config (see below) or sync your files?")
-      message('Sys.setenv(R_CONFIG_ACTIVE= "remote")')
-      check_host()
-    }
-    #only ever load the full data frame when absolutely necessary
-    if(all_genes){
-      wide_expression_data = suppressMessages(read_tsv(tidy_expression_file)) %>%
-        as.data.frame() %>%
-        pivot_wider(names_from = ensembl_gene_id, values_from = expression)
-      
-    }else{
-      # define gene ids and their type
-      if(!missing(hugo_symbols)){
-        gene_ids = hugo_symbols
-        gene_id_type = "Hugo_Symbol"
-        not_the_id_type = "ensembl_gene_id"
-      }else{
-        gene_ids = ensembl_gene_ids
-        gene_id_type = "ensembl_gene_id"
-        not_the_id_type = "Hugo_Symbol"
+  load_expression_by_samples = function(genes,samples,verbose){
+      split_dir = "/projects/rmorin/projects/gambl-repos/gambl-rmorin/results/icgc_dart/DESeq2-0.0_salmon-1.0/mrna--gambl-icgc-all/split"
```

Comment:
I definitely missed this update and the new setup. The grep  was returning gene expression data almost immediately, although I usually tried with just a handful of genes so don't know if it behaves same way on the large number of genes. Imagining the for loop approach is way slower than current engine setup but this was probably tested/benchmarked. Is this directory going to be also supported in the remote setup on local computer?
Separately, this path I think should be added to config instead of here

Reply:
Yes I need to move this to the config. I wanted to get confirmation from users of this function that the change was acceptable before I finalized such things. We would also need to automate the splitting of files by sample_id in order for this to continue working properly as new data is added to GAMBL. The performance improvements I was striving for here were mostly in relation to reducing the RAM footprint without slowing down the code. I think the code is actually faster now using grep but it would be interesting to compare. I could probably add another option to revert to the single file instead of scanning the per-sample files so we could do a head-to-head benchmark. What do you think?

Reply:
The per-sample split would be easy to do with additional snakemake rule and small script. I can bechmark different setups on GSC, and it might be good to have the option to revert to the single file - and see how it benchmarks, too